kernel/src/mem.rs

```diff
@@ -67,6 +65,26 @@ impl<L, T> AppSlice<L, T> {
     pub fn len(&self) -> usize {
         self.len
     }
+
+    pub unsafe fn ptr(&self) -> *const T {
+        self.ptr.ptr.get() as *const T
+    }
+
+    pub unsafe fn expose_to(&self, appid: AppId) -> bool {
+        let ps = &mut process::PROCS;
+        if appid.idx() != self.ptr.process.idx() && ps.len() > appid.idx() {
+            ps[appid.idx()]
+                .as_ref()
+                .map(|process| process.add_mpu_region(self.ptr() as *const u8, self.len()))
+                .unwrap_or(false)
+        } else {
+            false
+        }
+    }
+
+    pub fn iter(&self) -> slice::Iter<T> {
```

Comment:
You would think so, wouldn't you :P but it's not. Iterator is, but we don't need a special one so can just reuse the implementation for slices.